examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,351 @@
+Different conventions to define the focal mechanism are supported: Aki and Richards
+(``"aki"``), global CMT (``"gcmt"``), moment tensor (``"mt"``), partial focal
+mechanism (``"partial"``), and, principal axis (``"principal_axis"``). Please refer to
+the documentation of :meth:`pygmt.Figure.meca` regarding how to set up the input
+data in respect to the chosen input type and convention (i.e., the expected column
+order, keys, or column names).
+
+This tutorial focus on how to adjust the display of the beachballs:
+
+- Filling the quadrants
+- Plotting the components of a seismic moment tensor
+- Adjusting the outlines
+- Highlighting the nodal planes
+- Adding offset from the event location
+- Adding a label
+- Using size-coding and color-coding
+"""
+
+# %%
+import pandas as pd
+import pygmt
+
+# Set up arguments for basemap
+region = [-5, 5, -5, 5]
+projection = "X10c/4c"
+frame = ["af", "+ggray90"]
+
+
+# %%
+# Set up the focal mechanism data
+# -------------------------------
+#
+# Store focal mechanism parameters for one event in a dictionary following the
+
+# moment tensor convention
+mt_dict_single = {
+    "mrr": 4.71,
+    "mtt": 0.0381,
+    "mff": -4.74,
+    "mrt": 0.399,
+    "mrf": -0.805,
+    "mtf": -1.23,
+    "exponent": 24,
+}
+# Aki and Richards convention
+aki_dict_single = {"strike": 318, "dip": 89, "rake": -179, "magnitude": 7.75}
+
+
+# %%
+# Plotting a single beachball
+# ---------------------------
+#
+# Required parameters are ``spec` and ``scale`` as well as ``longitude``, ``latitude``
+# (event location), and depth (if these values are not included in the argument passed
+# to ``spec``). Additionally the ``convention`` parameter is required if ``spec`` is
+# an 1-D or 2-D numpy array; for the input types dictionary and ``pandas.Dataframe``,
+# ``convention`` is not needed.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(spec=mt_dict_single, scale="1c", longitude=0, latitude=0, depth=0)
+
+fig.show()
+
+
+# %%
+# Filling the quadrants
+# ---------------------
+#
+# Use the parameters ``compressionfill`` and ``extensionfill`` to fill the quadrants
+# with different colors or patterns. Regarding pattern see the gallery example
+# :doc:`Bit and hachure patterns </gallery/symbols/patterns>` and the Technical
+# Reference :doc:`Bit and hachure patterns </techref/patterns>`.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=mt_dict_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    compressionfill="darkorange",
+    extensionfill="cornsilk",
+)
+
+fig.meca(
+    spec=mt_dict_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="p8",
+    extensionfill="p31",
+    outline=True,
+)
+
+fig.show()
+
+
+# %%
+# Plotting the components of a seismic moment tensor
+# --------------------------------------------------
+#
```

Comment:
What about moving this section before the "Filling the quadrants" section? I.e., plot the expected beachballs first before tweaking the apperance.

Reply:
There was something I had in mind regarding the order of the sections; however, I am currently not remembering my idea here 🙈 (maybe this was somehow related to the example for **Fa**, **Fe**, **Fg**, **Ft**, **Fp**). As I think your suggestion makes sense, I move the "Filling the quadrants" section up in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/57121bfe702b4f1237317878b8278b454ad9653d.